bftengine/src/bftengine/SigManager.cpp

```diff
@@ -213,7 +213,12 @@ bool SigManager::verifySig(
     idOfReplica = replicasInfo_.isIdOfReplica(pid);
   }
   idOfReadOnlyReplica = replicasInfo_.isIdOfPeerRoReplica(pid);
-  ConcordAssert(idOfReplica || idOfExternalClient || idOfReadOnlyReplica);
```

Comment:
Asserts are usually in place to check programming assumptions. Why are you converting this to an exception? How is the exception handled at runtime?
A proper commit message would be helpful. What problem are you solving, why and how?

Reply:
@robem 
As discussed with @glevkovich, currently asserts are placed when we internally check the parameters.
But here pid is coming as a parameter in verifySig API
bool SigManager::verifySig(
    PrincipalId pid, const char* data, size_t dataLength, const char* sig, uint16_t sigLength)
 Therefore we have replaced assert with exception

Reply:
@robem , the input is comming from the network. Any byzantine replica may send bad data on message header and crash us. For the specific pid, I don't think it can ever happen since it is validated with tls ceritificate on the source on communication level. But we don't want to replay on external module (communication) - and what if we run tCP/udp?
So asserting here is not safe. replacing to exception.